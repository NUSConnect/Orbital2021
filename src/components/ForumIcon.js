import React from 'react'
import { Image, StyleSheet, Text, TouchableOpacity, Dimensions } from 'react-native'

const ForumIcon = ({ item, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container} testID='icon'>
      <Image
        style={styles.image}
        source={{
          uri:
                        item.forumImg ||
                        'https://firebasestorage.googleapis.com/v0/b/orbital2021-a4766.appspot.com/o/profile%2Fplaceholder.png?alt=media&token=8050b8f8-493f-4e12-8fe3-6f44bb544460'
        }}
        testID='image'
      />
      <Text style={styles.text} testID='text'>{item.forumName}</Text>
    </TouchableOpacity>
  )
}

export default ForumIcon

const styles = StyleSheet.create({
  container: {
    flex: 0,
    width: 130 * Dimensions.get('window').width / 414,
    height: 150,
    alignItems: 'center',
    justifyContent: 'center'
  },
  image: {
    width: 100,
    height: 100,
    borderRadius: 50,
    borderWidth: 4,
    borderColor: 'black'
  },
  text: {
    fontSize: 16,
    color: 'black'
  }
})
