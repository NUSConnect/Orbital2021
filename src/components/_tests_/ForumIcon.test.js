import * as React from "react";
import ForumIcon from "../ForumIcon";
import renderer from "react-test-renderer";

it(`renders correctly`, () => {
    const tree = renderer.create(<ForumIcon />);
    expect(tree).toMatchSnapshot();
});