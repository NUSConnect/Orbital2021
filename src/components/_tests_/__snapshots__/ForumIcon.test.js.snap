// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 0,
      "height": 150,
      "justifyContent": "center",
      "opacity": 1,
      "width": 130,
    }
  }
  testID="icon"
>
  <Image
    source={
      Object {
        "uri": "MockForumImg",
      }
    }
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 4,
        "height": 100,
        "width": 100,
      }
    }
    testID="image"
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
    testID="text"
  >
    TestName
  </Text>
</View>
`;
